.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateTimePeriodParser.cs

```diff
@@ -473,75 +448,143 @@ private DateTimeResolutionResult ParseSimpleCases(string text, DateObject refere
                     return ret;
                 }
 
-                // Parse "pm"
-                var pmStr = match.Groups[Constants.PmGroupName].Value;
-                var amStr = match.Groups[Constants.AmGroupName].Value;
-                var descStr = match.Groups[Constants.DescGroupName].Value;
-                if (!string.IsNullOrEmpty(amStr) || !string.IsNullOrEmpty(descStr) && descStr.StartsWith("a"))
-                {
-                    if (beginHour >= Constants.HalfDayHourCount)
-                    {
-                        beginHour -= Constants.HalfDayHourCount;
-                    }
+                var pastHours = endHour - beginHour;
```

Comment:
All timex generation code should be in timex generation, not spread throughout the code.